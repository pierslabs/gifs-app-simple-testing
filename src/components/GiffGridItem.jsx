import PropTypes from 'prop-types';

const GiffGridItem = ({ title, url }) => {
  return (
    <div className="col-md-auto">
      <div className="card p-2 m-2">
        <img
          src={url}
          className="rounded img-fluid"
          alt={title}
          style={{ height: 200 }}
        ></img>
        <a href={url} target="_blank" rel="noopener noreferrer">
          {title}
        </a>
      </div>
    </div>
  );
};

GiffGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GiffGridItem;
