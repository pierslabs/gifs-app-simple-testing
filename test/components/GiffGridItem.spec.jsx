import { getByRole, render, screen } from '@testing-library/react';
import GiffGridItem from '../../src/components/GiffGridItem';

describe('<GiffGridItem>', () => {
  const title = 'batman';
  const url = 'https://batman.com/batman.jpg';

  test('should match to snapshot', () => {
    const { container } = render(<GiffGridItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('should show the image with url and ALT ', () => {
    render(<GiffGridItem title={title} url={url} />);

    const { alt, src } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should show the title of component ', () => {
    render(<GiffGridItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
