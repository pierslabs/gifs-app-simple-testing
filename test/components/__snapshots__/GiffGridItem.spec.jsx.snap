// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiffGridItem> should match to snapshot 1`] = `
<div>
  <div
    class="col"
  >
    <div
      class="card p-2 m-2"
    >
      <img
        alt="batman"
        class="rounded img-fluid"
        src="https://batman.com/batman.jpg"
        style="height: 200px;"
      />
      <a
        href="https://batman.com/batman.jpg"
        rel="noopener noreferrer"
        target="_blank"
      >
        batman
      </a>
    </div>
  </div>
</div>
`;
